cpu/cc26x2_cc13x2/radio/rf_netdev.c

```diff
@@ -0,0 +1,439 @@
+         * an address */
+        eui64 = (uint8_t *)&FCFG->MAC_15_4_0;
+    }
+
+    /* The IEEE MAC address is stored in network byte order. */
+    for (i = 0; i < IEEE802154_LONG_ADDRESS_LEN; i++) {
+        out[i] = eui64[(IEEE802154_LONG_ADDRESS_LEN - 1) - i];
+    }
+}
+
+static int _init(netdev_t *dev)
+{
+    cc26x2_cc13x2_rf_netdev_t *netdev = (cc26x2_cc13x2_rf_netdev_t *)dev;
+
+    if (netdev == NULL) {
+        DEBUG_PUTS("_init: netdev is null!");
+        return -1;
+    }
+
+    /* Reset netdev_ieee802154 defaults */
+    netdev_ieee802154_reset(&netdev->netdev);
+
+    /* Set up EUI-64 address */
+    _read_device_ieee_eui64(netdev->netdev.long_addr);
+    memcpy(netdev->netdev.short_addr, netdev->netdev.long_addr + 6, 2);
+
+    /* https://tools.ietf.org/html/rfc4944#section-12
+     * Requires the first bit to 0 for unicast addresses */
+    netdev->netdev.short_addr[1] &= 0x7F;
+
+    /* Initialize data entries */
+    rfc_data_entry_gen_init(_rx_buf0, sizeof(_rx_buf0), sizeof(uint16_t), _rx_buf1);
+    rfc_data_entry_gen_init(_rx_buf1, sizeof(_rx_buf1), sizeof(uint16_t), _rx_buf2);
+    rfc_data_entry_gen_init(_rx_buf2, sizeof(_rx_buf2), sizeof(uint16_t), _rx_buf3);
+    rfc_data_entry_gen_init(_rx_buf3, sizeof(_rx_buf3), sizeof(uint16_t), _rx_buf0);
+
+    /* Initialize RX data queue */
+    rfc_data_queue_init(&_rx_queue, _rx_buf0);
+
+    /* Tune the radio to the given frequency */
+    rf_cmd_fs.frequency = 915;
+    rf_cmd_fs.fract_freq = 0;
+
+    /* Chain the CMD_FS command after running setup */
+    rf_cmd_prop_radio_div_setup.condition.rule = RFC_COND_ALWAYS;
+    rf_cmd_prop_radio_div_setup.condition.skip_no = 0;
+    rf_cmd_prop_radio_div_setup.next_op = (rfc_op_t *)&rf_cmd_fs;
+
+    /* Initialize radio driver in proprietary setup for the Sub-GHz band.
+     * To use Sub-GHz we need to use CMD_PROP_RADIO_DIV_SETUP as our setup
+     * command. */
+    rfc_init((rfc_op_t *)&rf_cmd_prop_radio_div_setup, rf_patch_cpe_prop,
+             _rfc_isr);
+
+    if (rfc_enable() < 0) {
+        DEBUG_PUTS("_init: rfc_enable failed!");
+        return -1;
+    }
+
+    return 0;
+}
+
+static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
+{
+    cc26x2_cc13x2_rf_netdev_t *dev = (cc26x2_cc13x2_rf_netdev_t *)netdev;
+    int res = -ENOTSUP;
+
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+
+    switch (opt) {
+        case NETOPT_RX_END_IRQ:
+            if (len != sizeof(bool)) {
+                return -EINVAL;
+            }
+            else {
+                if (*(const bool *)val) {
+                    RFC_DBELL_NONBUF->RFCPEIEN |= CPE_IRQ_RX_ENTRY_DONE;
+                }
+                else {
+                    RFC_DBELL_NONBUF->RFCPEIEN &= ~CPE_IRQ_RX_ENTRY_DONE;
+                }
+            }
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_TX_END_IRQ:
+            if (len != sizeof(bool)) {
```

Comment:
```suggestion
            if (len != sizeof(netopt_enable_t)) {
```

Reply:
I've changed it, maybe using `netopt_enable_t` in `_tx_end_irq` could be better?